### PR TITLE
Asset Manager

### PR DESCRIPTION
**Pull request by NOOBDY**

### CODE REVIEW

**Review by Dragonfly911117 — DISMISSED** [review]
image

Significant optimization(on 10k Giraffes)!

File: include/Util/Text.hpp [hunk 1]
```diff
@@ -74,6 +74,8 @@ class Text : public Core::Drawable {
     static std::unique_ptr<Core::VertexArray> s_VertexArray;
     static std::unique_ptr<Core::UniformBuffer<Core::Matrices>> s_UniformBuffer;
 
+    static Util::AssetStore<std::shared_ptr<SDL_RWops>> s_Store;
```
Comment (ntut-rick) [inline include/Util/Text.hpp:77]: Maybe adding a note explaining that this is for `TTF_OpenFontRW`

**Review by ntut-rick — DISMISSED** [review]
LGTM

**Review by ntut-rick — DISMISSED** [review]
OK.

File: include/Util/AssetStore.inl [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+template <typename T>
+void AssetStore<T>::Load(const std::string &filepath) {
+    m_Map[filepath] = m_Loader(filepath);
+}
+
+template <typename T>
+T AssetStore<T>::Get(const std::string &filepath) {
+    auto result = m_Map.find(filepath);
+    if (result != m_Map.end()) {
+        return result;
```
[the remark this answers is not in the document]
Comment (NOOBDY) [reply include/Util/AssetStore.inl]: fixed in feaad8c

[the remark this answers is not in the document]
Comment (NOOBDY) [reply include/Util/AssetStore.inl]: fixed in feaad8c

[the remark this answers is not in the document]
Comment (NOOBDY) [reply include/Util/AssetStore.inl]: fixed in feaad8c

[the remark this answers is not in the document]
Comment (NOOBDY) [reply include/Util/AssetStore.inl]: fixed in feaad8c

**Review by Dragonfly911117 — APPROVED** [review]

**Pull request merged**